### PR TITLE
Core Data: Adds 'include' to the query key

### LINKED ISSUE #34506: Error on replace image in Post Featured Image block
## Description
If I use the featured image block and try to replace the image with another image, the new image will not be rendered and I get a warning in the error console

## Step-by-step reproduction instructions
- Add `Post Featured Image` block
- Select image
- Click on `Replace` and select another image

No image is rendered and a warning will be in the error console 

```
react-dom.development.js:61 
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    at MediaReplaceFlow (/gutenberg/build/block-editor/index.min.js?ver=893f09019a5c32d5f4b49f064ac0e4c2:28350:3)
    at FilteredComponentRenderer (/gutenberg/build/components/index.min.js?ver=a610429ef1f11ac7a26bcdd28c16089b:72671:9)
    at /gutenberg/build/data/index.min.js?ver=ba46eec1abdc965327bf5d614463102d:2004:79
    at Fill (/gutenberg/build/components/index.min.js?ver=a610429ef1f11ac7a26bcdd28c16089b:81842:3)
    at Fill
    at BlockControlsOtherFill
    at StyleProvider (/gutenberg/build/components/index.min.js?ver=a610429ef1f11ac7a26bcdd28c16089b:83170:3)
    at BlockControlsFill (/gutenberg/build/block-editor/index.min.js?ver=893f09019a5c32d5f4b49f064ac0e4c2:8407:3)
```

### PR DESCRIPTION
**Pull request by Mamaduka**
## Description
Adds `include` argument to the query key (`stableKey`). Trying to use `itemIds` from the default stable key cache turned out to be problematic.

TL;DR - Two requests with different include values cannot have the same results.
Details - https://github.com/WordPress/gutenberg/pull/34034#issuecomment-913414279.

P.S. Should we consider a different "hashing" method for `stableKey`. For example, querying with `include` array size of 100 would result in key something like this:

```
include=0%2C1%2C2%2C3%2C4%2C5%2C6%2C7%2C8%2C9%2C10%2C11%2C12%2C13%2C14%2C15%2C16%2C17%2C18%2C19%2C20%2C21%2C22%2C23%2C24%2C25%2C26%2C27%2C28%2C29%2C30%2C31%2C32%2C33%2C34%2C35%2C36%2C37%2C38%2C39%2C40%2C41%2C42%2C43%2C44%2C45%2C46%2C47%2C48%2C49%2C50%2C51%2C52%2C53%2C54%2C55%2C56%2C57%2C58%2C59%2C60%2C61%2C62%2C63%2C64%2C65%2C66%2C67%2C68%2C69%2C70%2C71%2C72%2C73%2C74%2C75%2C76%2C77%2C78%2C79%2C80%2C81%2C82%2C83%2C84%2C85%2C86%2C87%2C88%2C89%2C90%2C91%2C92%2C93%2C94%2C95%2C96%2C97%2C98%2C99
```

Fixes #34506.

## How has this been tested?
Unit tests should be passing.

Can be tested in browser console using following snippet
```js
// Triggers resolver and REST API request
wp.data.select('core').getMedia( 1, { context: 'view' } );

// Also triggers resolver  and REST API request
wp.data.select('core').getMedia( 2, { context: 'view' } );

// Gets item from cache
wp.data.select('core').getMedia( 1, { context: 'view' } );
```

## Types of changes
Bugfix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by Mamaduka** [comment]
I think this also fixes #32594. I'm no longer able to reproduce the issue using the steps described here - https://github.com/WordPress/gutenberg/issues/32594#issuecomment-913004360.

But I think the issue with `getMergedItemIds` remains if the method should return unique IDs.

**Comment by youknowriad** [comment]
This looks good to me.

> But I think the issue withgetMergedItemIds remains if the method should return unique IDs.

Do you mind clarifying a bit more here, I'm not sure what this refers to. Do you know why the "include" was excluded from the key initially (I guess maybe related to smart behavior we removed in the previous PR)

**Comment by Mamaduka** [comment]
Sorry for the confusion.

Comment about `getMergedItemIds` is related to this issue - https://github.com/WordPress/gutenberg/issues/32594#issuecomment-913004360. This change also fixes it.

I tracked down the original problem to the `getMergedItemIds` and just wanted to make sure that current behavior, not checking if IDs are unique, was correct.

The `include` was introduced alongside the `_fields` param data reuse; PR has no additional information about it - #19498.

But I think you're right. It was excluded because of the smart logic we removed in the previous PR.

**Review by youknowriad — APPROVED** [review]
I think this makes a lot of sense to me. 

Obviously these internals are pretty impactful and sometimes fragile. I wonder if there's a way for us to provide a test that avoid regression on a higher-level (not sure how though)

**Comment by Mamaduka** [comment]
Thanks again for the review and feedback, @youknowriad 

> I wonder if there's a way for us to provide a test that avoid regression on a higher-level (not sure how though).

Agree. This is worth exploring. I'll what I can do in the near future.
